docs/readme/README.jl

```diff
@@ -0,0 +1,247 @@
+
+using LinearAlgebra
+
+tilerange(ti, ts) = (ti-1)*ts+1:ti*ts
+
+function cholesky_tiled!(A, ts)
+    m = size(A, 1); @assert m==size(A, 2)
+    m%ts != 0 && error("Tilesize doesn't fit the matrix")
+    n = m÷ts  # number of tiles in each dimension
+
+    T = [view(A, tilerange(i, ts), tilerange(j, ts)) for i in 1:n, j in 1:n]
+
+    for i in 1:n
+        ## Diagonal cholesky serial factorization
+        cholesky!(T[i,i])
+
+        ## Left blocks update
+        U = UpperTriangular(T[i,i])
+        for j in i+1:n
+            ldiv!(U', T[i,j])
+        end
+
+        ## Submatrix update
+        for j in i+1:n
+            for k in j:n
+                mul!(T[j,k], T[i,j]', T[i,k], -1, 1)
+            end
+        end
+    end
+
+    ## Construct the factorized object
+    return Cholesky(A, 'U', zero(LinearAlgebra.BlasInt))
+end
+
+# Parallelizing the code with `DataFlowTasks.jl` is as easy as wrapping function calls within `@dspawn`, and adding annotations describing data access modes:
+
+using DataFlowTasks
+
+function cholesky_dft!(A, ts)
+    m = size(A, 1); @assert m==size(A, 2)
+    m%ts != 0 && error("Tilesize doesn't fit the matrix")
+    n = m÷ts  # number of tiles in each dimension
+
+    T = [view(A, tilerange(i, ts), tilerange(j, ts)) for i in 1:n, j in 1:n]
+
+    for i in 1:n
+        ## Diagonal cholesky serial factorization
+        @dspawn cholesky!(@RW(T[i,i])) label="chol ($i,$i)"
+
+        ## Left blocks update
+        U = UpperTriangular(T[i,i])
+        for j in i+1:n
+            @dspawn ldiv!(@R(U)', @RW(T[i,j])) label="ldiv ($i,$j)"
+        end
+
+        ## Submatrix update
+        for j in i+1:n
+            for k in j:n
+                @dspawn mul!(@RW(T[j,k]), @R(T[i,j])', @R(T[i,k]), -1, 1) label="schur ($j,$k)"
+            end
+        end
+    end
+
+    ## Construct the factorized object
+    r = @dspawn Cholesky(@R(A), 'U', zero(LinearAlgebra.BlasInt)) label="result"
+    return fetch(r)
+end
+
+# (Also note how extra annotations were added in the code, in order to attach meaningful labels to the tasks. These will later be useful to interpret the output of debugging & profiling tools.)
+
+#-
+
+# The code below shows how to use this `cholesky_tiled!` function, as well as how to profile the program and get information about how tasks were scheduled:
+
+## DataFlowTasks environnement setup
+DataFlowTasks.reset!()
+DataFlowTasks.enable_log()
+
+## Context
+n  = 2048
+ts = 512
+A = rand(n, n)
+A = (A + adjoint(A))/2
+A = A + n*I;
+
+#-
+@info "Testing sequential Cholesky factorization"          #hide
+F = cholesky_tiled!(copy(A), ts)                           #hide
+                                                           #hide
+## Check results                                           #hide
+err = norm(F.L*F.U-A,Inf)/max(norm(A),norm(F.L*F.U))       #hide
+@show err                                                  #hide
+@assert err < eps(Float64)                                 #hide
```

Comment:
Naive question: why not something simpler, like
```
err = norm(F.L*F.U - A) / norm(A)
```
Also: I arbitrarily put the tolerance to machine precision, but this might not be the right thing to do.